kpops/pipeline.py

```diff
@@ -205,19 +203,19 @@ def __add_input(self, topic_id: str, target: str) -> None:
 
 
 def create_env_components_index(
-    environment_components: list[dict],
-) -> dict[str, dict]:
+    environment_components: list[dict[str, Any]],
+) -> dict[str, dict[str, Any]]:
     """Create an index for all registered components in the project.
 
     :param environment_components: List of all components to be included
     :return: component index
     """
-    index: dict[str, dict] = {}
+    index: dict[str, dict[str, Any]] = {}
     for component in environment_components:
         if "type" not in component or "name" not in component:
             msg = "To override components per environment, every component should at least have a type and a name."
             raise ValueError(msg)
-        index[component["name"]] = component
```

Comment:
@disrupted What does this TODO mean?

Reply:
we want to access components by their `id` from the index rather than by their name. This depends on the topics refactor I prepared in the other PR